packages/x-tree-view/src/internals/useTreeView/useTreeView.ts

```diff
@@ -88,42 +89,46 @@ export const useTreeView = <Plugins extends readonly TreeViewPlugin<TreeViewAnyP
 
   plugins.forEach(runPlugin);
 
-  contextValue.runItemPlugins = ({ props, ref }) => {
-    let finalProps = props;
-    let finalRef = ref;
-    const itemWrappers: ((children: React.ReactNode) => React.ReactNode)[] = [];
+  contextValue.runItemPlugins = (itemPluginProps) => {
+    let finalRootRef: React.RefCallback<HTMLLIElement> | null = null;
+    let finalContentRef: React.RefCallback<HTMLElement> | null = null;
 
     plugins.forEach((plugin) => {
       if (!plugin.itemPlugin) {
         return;
       }
 
-      const itemPluginResponse = plugin.itemPlugin({ props: finalProps, ref: finalRef });
-      if (itemPluginResponse?.props) {
-        finalProps = itemPluginResponse.props;
+      const itemPluginResponse = plugin.itemPlugin({
+        props: itemPluginProps,
+        rootRef: finalRootRef,
+        contentRef: finalContentRef,
+      });
+      if (itemPluginResponse?.rootRef) {
+        finalRootRef = itemPluginResponse.rootRef;
       }
-      if (itemPluginResponse?.ref) {
-        finalRef = itemPluginResponse.ref;
-      }
-      if (itemPluginResponse?.wrapItem) {
-        itemWrappers.push(itemPluginResponse.wrapItem);
+      if (itemPluginResponse?.contentRef) {
+        finalContentRef = itemPluginResponse.contentRef;
       }
     });
 
     return {
-      props: finalProps,
-      ref: finalRef,
-      wrapItem: (children) => {
-        let finalChildren: React.ReactNode = children;
-        itemWrappers.forEach((itemWrapper) => {
-          finalChildren = itemWrapper(finalChildren);
-        });
-
-        return finalChildren;
-      },
+      contentRef: finalContentRef,
+      rootRef: finalRootRef,
     };
   };
 
+  const itemWrappers = plugins
+    .map((plugin) => plugin.wrapItem)
+    .filter((wrapItem): wrapItem is TreeItemWrapper => !!wrapItem);
+  contextValue.wrapItem = ({ nodeId, children }) => {
```

Comment:
I moved `wrapItem` outside of `runItemPlugins` / `itemPlugin` so that the `TreeItemProvider` can wrap items without executing `runItemPlugins`.

Before, we executed `runItemPlugins` in `TreeItem` and then used the returned `wrapItem` function to wrap the JSX element.
But when doing a custom field, it's a weird DX to do:

```tsx
const node = (
  <Root>
    <Content>STUFF</Content>
  </Root>
);

return wrapItem(node);
```

So I replaced it with: 

```tsx
return (
  <TreeItemProvider nodeId={nodeId}r>
    <Root>
      <Content>STUFF</Content>
    </Root>
);
```

Which means `TreeItemProvider` is not aware of the value returned by `runItemPlugins`, thus the split.